app/services/authorizer.rb

```diff
@@ -19,9 +19,26 @@ def can?(permission, subject = nil)
   end
 
   def find_collection(resource_class, options = {})
+    filter_limits = nil
+    user.filters.each do |filter|
+      if filter.filters.length > 0
+        results = find_filter_collection(filter.filters, resource_class, options)
+        filter_limits = filter_limits.nil? ? results : filter_limits.merge(results)
+      end
+    end
+    filter_collection = find_filter_collection(user.filters, resource_class, options)
+    if filter_limits.nil?
+      filter_collection
+    else
+      ids = (filter_limits & filter_collection).collect { |o| o.id }
+      resource_class.where(:id => ids)
+    end
+  end
+
```

Comment:
I think that this method should be rather implemented by composing of limiting filters and user's own filter and searching for collection at once, something like `(filter_limits condition) AND (own filters condition)`